### ISSUE TITLE
add batch features to virtual jdbc layer

### ISSUE THREAD

**Issue by NathanQingyangXu** [NathanQingyangXu/jpa-mongodb-mapping#8]

**Comment by jyemin** [comment]
Is this about properly implementing `MongoPreparedStatement#addBatch` so that `hibernate.jdbc.batch_size` will work as expected?

**Comment by NathanQingyangXu** [comment]
Might be. More concern is Hibernate's internal batch feature, but not sure
about whether ultimately we need to change this jdbc method.

Please refer to
https://docs.jboss.org/hibernate/orm/6.5/userguide/html_single/Hibernate_User_Guide.html#batch

A common perf trick in JPA, but we can delay it later.

**Comment by jyemin** [comment]
I tried setting `hibernate.jdbc.batch_size ` to 2 and persisting 2 documents to the same table.  It definitely calls `addBatch` in that scenario.

**Comment by NathanQingyangXu** [comment]
Yeah, that is as expected. It also aligns with our json command accepting
multiple docs. So it seems a good feature to have!
